design/design.md

```diff
@@ -162,21 +166,36 @@ for Zcash script inspection, debugging, etc.
 
 ### Internal Dependencies
 
-- `zebra-chain`
-- `zebra-state`
-- `zebra-script`
+- `zebra-chain` for data structures and parsing.
+- `zebra-state` to read and update the state database.
+- `zebra-script` for script parsing and validation.
 
 ### Responsible for
 
 - consensus-specific parameters (network magics, genesis block, pow
   parameters, etc) that determine the network consensus
 - consensus logic to decide which block is the current block
-- all context-dependent validation logic, e.g., determining whether a
-  transaction is accepted in a particular chain state context.
+- block and transaction validation
+  - context-free validation, e.g., signature, proof verification, etc.
+  - context-dependent validation, e.g., determining whether a
+    transaction is accepted in a particular chain state context.
+- block checkpoints
+  - mandatory checkpoints (genesis block, sapling activation)
+  - optional regular checkpoints (every Nth block)
+- modifying the chain state
+  - adding new blocks to `ZebraState`
+  - adding new transactions to `ZebraMempool`
+- providing `tower::Service` interfaces for all of the above to
+  support backpressure and batch validation.
 
 ### Exported types
 
-- [...]
+- ZebraStateService
```

Comment:
I'm not sure how we should describe it in the `design.md`, but one thing that will likely end up happening is that rather than exporting a concrete type that implements `Service`, we would export a function that returns `impl Service<...>` with some generic bounds, as is done in `zebra-network`.

The reason is that we want to be able to make use of Tower layers (e.g., using a `Buffer` to make the service `Clone`, or adding metrics,  or a retry policy, or timeouts, or...), and those work by wrapping an inner service type in another wrapper type that also implements `Service`.  Returning `impl Service<...>` means that we are free to return a `Buffer<ZebraState>` or a `Buffer<Retry<Timeout<PeakEwma<...>>>>` or whatever combination of layers we want.

Reply:
Fixed in dec5c37fb5bc08eab967064ca02069848ec117ff so that it's similar to the zebra-network service docs.